lib/logdna/client.rb

```diff
@@ -11,153 +12,114 @@ def initialize(request, uri, opts)
-      write_to_buffer(next_object[:msg], next_object[:opts])
-    end
+      if @lock.try_lock
+          if !@side_messages.empty?
+            @buffer.concat(@side_messages)
+          end
+          processed_message = process_message(msg, opts)
+          new_message_size = processed_message.to_s.bytesize
+          @buffer_byte_size += new_message_size
+
+          if @flush_limit > (new_message_size + @buffer_byte_size)
+             @buffer.push(processed_message)
+          else
+             @buffer.push(processed_message)
+             self.flush
+          end
 
-    def process_buffer(buffer_size)
-      if buffer_size > @actual_byte_limit
-        @buffer_over_limit = true
-        flush()
-        @buffer_over_limit = false
+          begin
+            @lock.unlock
+          rescue
+            puts 'Nothing was locked'
+          end
+          schedule_flush()
       else
-        @task = create_flush_task
+          @side_messages.push(process_message(msg, opts))
       end
     end
 
-    # this should be running synchronously if @buffer_over_limit i.e. called from self.buffer
-    # else asynchronously through @task
-    def flush()
-      if defined? @@request and !@@request.nil?
-        request_messages = []
-        @lock.synchronize do
-          request_messages = @messages
-          @buffer.truncate(0)
-          @messages = []
-        end
-        return if request_messages.empty?
-
-        real = {
+    def flush
+      return if @buffer.empty?
+      if @lock.try_lock
+        @request.body = {
           e: 'ls',
-          ls: request_messages,
+          ls: @buffer.concat(@side_messages),
         }.to_json
+        @timer_task = false
+        @side_messages.clear
 
-        @@request.body = real
-        @response = Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: @uri.scheme == 'https') do |http|
-          http.request(@@request)
+        begin
+          @response = Net::HTTP.start(@uri.hostname, @uri.port, use_ssl: @uri.scheme == 'https') do |http|
+            http.request(@request)
+          end
+          @exception_flag = false
+        rescue
+          puts `Error at the attempt to send the request #{@response.body if @response}`
+          @exception_flag = true
+          @side_messages.concat(@buffer)
         end
+        @buffer.clear
 
-        # don't kill @task if this was executed from self.buffer
-        # don't kill @task if there are queued messages
-        unless @buffer_over_limit || @messages.any? || @task.nil?
-          @task.shutdown
-          @task.kill
+        begin
+          @lock.unlock
+        rescue
+          puts 'Nothing was locked'
         end
       end
-    end
+   end
 
-    def exitout()
-      check_side_buffer
-      if @messages.any?
+   def exitout
+      if @buffer.any?
         flush()
       end
-      puts "Exiting LogDNA logger: Logging remaining messages"
+        puts "Exiting LogDNA logger: Logging remaining messages"
```

Comment:
This indentation looks wrong. Did you lint this file? Try Rubocop if you haven't.